eng/scripts/Update-VcpkgNightlyBranch.ps1

```diff
@@ -0,0 +1,75 @@
+<#
+.SYNOPSIS
+Updates a Target branch with changes from a Main branch and a Source branch
+
+.DESCRIPTION
+Updat the long-lived nightly branch with changes from Main and a given Source 
+branch. Changes are first merged from the Main branch and then from the Source 
+branch. In both cases conflicts are resolved in favor of the branch which is 
+being merged.
+
+.PARAMETER SourceBranchName
+Name of the branch with new changes to merge into the Target Branch
+
+.PARAMETER TargetBranchName
+Name of the long-lived branch receiving the changes
+
+.PARAMETER MainBranchName
+Name of the "Main" branch to merge into the Target Branch
+
+#>
+
+param(
+    [string] $SourceBranchName,
+    [string] $TargetBranchName,
+    [string] $MainBranchName
+)
+
+$ErrorActionPreference = "Continue"
+
+if ((Get-Command git | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate git. Install git https://git-scm.com/downloads"
+    exit 1
+}
+
+Write-Host "git fetch --all"
+git fetch --all
+
+# Fail quickly if the $SourceBranchName of nightly changes cannot be found
+Write-Host "git checkout $SourceBranchName"
+git checkout $SourceBranchName
+
+if ($LASTEXITCODE -ne 0) { 
+    Write-Error "Cannot find branch: $SourceBranchName, nothing to merge"
+    exit 1
+}
+
+
+Write-Host "git branch --show-current"
+$currentBranch = git branch --show-current
+
+if ($currentBranch -ne $TargetBranchName) {
```

Comment:
Have a look at the logic Sima and I recently started using at https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/git-branch-push.ps1#L65 to enable checking out a branch that may or may not exist locally.